app/src/main/java/com/glia/exampleapp/Application.java

```diff
@@ -15,6 +15,7 @@ public class Application extends android.app.Application {
     public void onCreate() {
         super.onCreate();
         GliaWidgets.onAppCreate(this);
+        GliaWidgets.setCustomMessageAdapter(new ExampleCustomMessageAdapter());
```

Comment:
May it be better to set the message adapter after Glia.init()? To me it's more related to initialization, Glia setup.

Reply:
Not sure. Actually, the adapter can be re-setup.